lib/driver.js

```diff
@@ -264,8 +264,13 @@ class XCUITestDriver extends BaseDriver {
     this.logEvent('appConfigured');
 
     // fail very early if the app doesn't actually exist
+    // or if bundle id doesn't point to an installed app
     if (this.opts.app) {
       await checkAppPresent(this.opts.app);
+    } else if (this.opts.bundleId && !this.safari) {
```

Comment:
Well... to my mind they are different types of conditions, and ought to be separated. The outer is all about the opts, while the inner is about the device.